src/Route.php

```diff
@@ -170,10 +174,31 @@ public function defaults(array $defaults): self
 
     /**
      * @param callable|MiddlewareInterface $middleware
-     * @return MiddlewareInterface
+     * @return MiddlewareInterface|string|array
      */
-    private function prepareMiddleware($middleware): MiddlewareInterface
+    private function prepareMiddleware($middleware)
     {
+        if (
+            is_string($middleware) && class_exists($middleware)
+            && in_array(MiddlewareInterface::class, class_implements($middleware))
+        ) {
+            if ($this->container === null) {
+                throw new InvalidArgumentException('Route container must not be null for lazy loaded middleware.');
+            }
+            return $middleware;
+        }
+
+        if (
+            is_array($middleware) && count($middleware) === 2
+            && isset($middleware[0]) && isset($middleware[1])
```

Comment:
it can be `isset($middleware[0], $middleware[1])` or `array_keys($middleware) === [0, 1]`

Reply:
I think this check is not needed because the next step is checked for a string

Reply:
`array_keys($middleware) === [0, 1]` can replace `count($middleware) === 2 && isset($middleware[0]) && isset($middleware[1])`

> because the next step is checked for a string

if i pass array like `['middleware' => SomeClass::class, 'method' => 'index']` then next step will not work correctly

Reply:
> `array_keys($middleware) === [0, 1]` can replace `count($middleware) === 2 && isset($middleware[0]) && isset($middleware[1])`

Is it faster?

Reply:
> Is it faster?

¯\\\_(ツ)\_/¯